frontend/scripts/react-components/tool-links/tool-links.reducer.js

```diff
@@ -142,22 +141,9 @@ const toolLinksReducer = {
       // TODO: if any selectedNode, make those columns visible (selected)
     });
   },
+  [TOOL_LINKS__SET_NODES]: setNodes,
 
-  [TOOL_LINKS__SET_NODES](state, action) {
-    const { nodes } = action.payload;
-    return immer(state, draft => {
-      draft.data.nodes = {};
```

Comment:
My only concern is that this gets triggered everytime we change context, and if we use only one action we need a way of deleting the existing nodes.

Reply:
Yeah, good one. But shouldn't we store the links and nodes by context so we don't have to retrieve them again every time we change the context?

Reply:
The browser will cache them, so we don't need it. And that way we avoid memory leaks.